core/src/test/kotlin/io/specmatic/core/CalculatePathTest.kt

```diff
@@ -726,4 +726,205 @@ internal class CalculatePathTest {
+            pattern = mapOf(
+                "level2" to level2Pattern
+            ),
+            typeAlias = "Level1Object"
+        )
+
+        val value = JSONObjectValue(mapOf(
+            "level2" to JSONObjectValue(mapOf(
+                "level3" to JSONObjectValue(mapOf(
+                    "data" to StringValue("test")
+                ))
+            ))
+        ))
+
+        val paths = level1Pattern.calculatePath(value, Resolver())
+        
+        assertThat(paths).containsExactly("{Level1Object}.level2.{Level2Object}.level3.{Level3Object}.data{string}")
+    }
+    
+    @Test
+    fun `calculatePath should handle nested AnyPattern containing JSONObjectPattern`() {
+        // Test case 2: Nested structure with AnyPattern containing JSONObjectPattern
+        val innerObjectPattern = JSONObjectPattern(
+            pattern = mapOf(
+                "innerData" to StringPattern()
+            ),
+            typeAlias = "InnerObject"
+        )
+        
+        val pattern = JSONObjectPattern(
+            pattern = mapOf(
+                "field" to AnyPattern(listOf(
+                    innerObjectPattern,
+                    StringPattern()
+                ))
+            ),
+            typeAlias = "OuterObject"
+        )
+
+        val value = JSONObjectValue(mapOf(
+            "field" to JSONObjectValue(mapOf(
+                "innerData" to StringValue("test")
+            ))
+        ))
+
+        val paths = pattern.calculatePath(value, Resolver())
+        
+        assertThat(paths).containsExactly("{OuterObject}.field{InnerObject}")
+    }
+    
+    @Test
+    fun `calculatePath should handle nested AnyPattern with DeferredPattern`() {
+        // Test case 3: Nested structure using DeferredPattern without direct typeAlias
+        val patterns = mapOf(
+            "(NestedData)" to JSONObjectPattern(
+                pattern = mapOf(
+                    "value" to StringPattern()
+                )
+            )
+        )
+        
+        val pattern = JSONObjectPattern(
+            pattern = mapOf(
+                "level1" to JSONObjectPattern(
+                    pattern = mapOf(
+                        "level2" to AnyPattern(listOf(
+                            DeferredPattern("(NestedData)"),
+                            NumberPattern()
+                        ))
+                    ),
+                    typeAlias = "MiddleLevel"
+                )
+            ),
+            typeAlias = "TopLevel"
+        )
+
+        val value = JSONObjectValue(mapOf(
+            "level1" to JSONObjectValue(mapOf(
+                "level2" to JSONObjectValue(mapOf(
+                    "value" to StringValue("test")
+                ))
+            ))
+        ))
+
+        val paths = pattern.calculatePath(value, Resolver(newPatterns = patterns))
+        
+        assertThat(paths).containsExactly("{TopLevel}.level1.{MiddleLevel}.level2{NestedData}")
```

Comment:
This path set should have contained: "{TopLevel}.level1{MiddleLevel}.level2{NestedData}".

Update the test accordingly, and then update the application code to pass the test.